### ISSUE TITLE
Incorrect ordering of x-labels in autoplot.BenchmarkResult

### ISSUE THREAD

**Issue by RaphaelS1** [mlr-org/mlr3viz#53]
``` r
library(mlr3learners)
library(mlr3viz)
library(mlr3)
task = tsk("boston_housing")

learners = lrns(c("regr.featureless", "regr.rpart", "regr.ranger"))
measure = msr("regr.rmse")
set.seed(1)
bmr = benchmark(benchmark_grid(task, learners, rsmp("cv", folds = 3)))
bmr$score(measure)[,c("learner_id", "regr.rmse")]
#>          learner_id regr.rmse
#> 1: regr.featureless  8.455316
#> 2: regr.featureless 10.222279
#> 3: regr.featureless  8.974639
#> 4:       regr.rpart  1.729365
#> 5:       regr.rpart  1.811437
#> 6:       regr.rpart  1.608334
#> 7:      regr.ranger  1.636241
#> 8:      regr.ranger  2.125320
#> 9:      regr.ranger  1.901388
mlr3viz:::autoplot.BenchmarkResult(bmr, measure = measure)
```

![](https://i.imgur.com/LOb27vO.png)

<sup>Created on 2020-10-01 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.2 (2020-06-22)
#>  os       macOS Catalina 10.15.3      
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_GB.UTF-8                 
#>  ctype    en_GB.UTF-8                 
#>  tz       Europe/London               
#>  date     2020-10-01                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package      * version    date       lib source                               
#>  assertthat     0.2.1      2019-03-21 [1] CRAN (R 4.0.0)                       
#>  backports      1.1.9      2020-08-24 [1] CRAN (R 4.0.2)                       
#>  callr          3.4.4      2020-09-07 [1] CRAN (R 4.0.2)                       
#>  checkmate      2.0.0      2020-02-06 [1] CRAN (R 4.0.0)                       
#>  cli            2.0.2      2020-02-28 [1] CRAN (R 4.0.0)                       
#>  codetools      0.2-16     2018-12-24 [1] CRAN (R 4.0.2)                       
#>  colorspace     1.4-1      2019-03-18 [1] CRAN (R 4.0.0)                       
#>  crayon         1.3.4      2017-09-16 [1] CRAN (R 4.0.0)                       
#>  curl           4.3        2019-12-02 [1] CRAN (R 4.0.0)                       
#>  data.table     1.13.0     2020-07-24 [1] CRAN (R 4.0.2)                       
#>  desc           1.2.0      2018-05-01 [1] CRAN (R 4.0.0)                       
#>  devtools       2.3.0      2020-04-10 [1] CRAN (R 4.0.0)                       
#>  digest         0.6.25     2020-02-23 [1] CRAN (R 4.0.0)                       
#>  dplyr          1.0.2      2020-08-18 [1] CRAN (R 4.0.2)                       
#>  ellipsis       0.3.1      2020-05-15 [1] CRAN (R 4.0.2)                       
#>  evaluate       0.14       2019-05-28 [1] CRAN (R 4.0.0)                       
#>  fansi          0.4.1      2020-01-08 [1] CRAN (R 4.0.0)                       
#>  farver         2.0.3      2020-01-16 [1] CRAN (R 4.0.0)                       
#>  fs             1.4.1      2020-04-04 [1] CRAN (R 4.0.0)                       
#>  future         1.18.0     2020-07-09 [1] CRAN (R 4.0.2)                       
#>  future.apply   1.6.0      2020-07-01 [1] CRAN (R 4.0.2)                       
#>  generics       0.0.2      2018-11-29 [1] CRAN (R 4.0.0)                       
#>  ggplot2        3.3.2      2020-06-19 [1] CRAN (R 4.0.2)                       
#>  globals        0.12.5     2019-12-07 [1] CRAN (R 4.0.0)                       
#>  glue           1.4.2      2020-08-27 [1] CRAN (R 4.0.2)                       
#>  gtable         0.3.0      2019-03-25 [1] CRAN (R 4.0.0)                       
#>  highr          0.8        2019-03-20 [1] CRAN (R 4.0.0)                       
#>  htmltools      0.4.0      2019-10-04 [1] CRAN (R 4.0.0)                       
#>  httr           1.4.1      2019-08-05 [1] CRAN (R 4.0.0)                       
#>  knitr          1.29       2020-06-23 [1] CRAN (R 4.0.2)                       
#>  labeling       0.3        2014-08-23 [1] CRAN (R 4.0.0)                       
#>  lattice        0.20-41    2020-04-02 [1] CRAN (R 4.0.2)                       
#>  lgr            0.3.4      2020-03-20 [1] CRAN (R 4.0.0)                       
#>  lifecycle      0.2.0      2020-03-06 [1] CRAN (R 4.0.0)                       
#>  listenv        0.8.0      2019-12-05 [1] CRAN (R 4.0.0)                       
#>  magrittr       1.5        2014-11-22 [1] CRAN (R 4.0.0)                       
#>  Matrix         1.2-18     2019-11-27 [1] CRAN (R 4.0.2)                       
#>  memoise        1.1.0      2017-04-21 [1] CRAN (R 4.0.0)                       
#>  mime           0.9        2020-02-04 [1] CRAN (R 4.0.0)                       
#>  mlr3         * 0.6.0-9000 2020-09-14 [1] Github (mlr-org/mlr3@f7fb636)        
#>  mlr3learners * 0.3.0      2020-09-12 [1] Github (mlr-org/mlr3learners@50b4169)
#>  mlr3measures   0.2.0      2020-06-27 [1] CRAN (R 4.0.2)                       
#>  mlr3misc       0.5.0      2020-08-13 [1] CRAN (R 4.0.2)                       
#>  mlr3viz      * 0.2.0-9000 2020-09-10 [1] Github (mlr-org/mlr3viz@8d1f98c)     
#>  munsell        0.5.0      2018-06-12 [1] CRAN (R 4.0.0)                       
#>  paradox        0.4.0-9000 2020-09-09 [1] Github (mlr-org/paradox@e10d740)     
#>  pillar         1.4.6      2020-07-10 [1] CRAN (R 4.0.2)                       
#>  pkgbuild       1.1.0      2020-07-13 [1] CRAN (R 4.0.2)                       
#>  pkgconfig      2.0.3      2019-09-22 [1] CRAN (R 4.0.0)                       
#>  pkgload        1.1.0      2020-05-29 [1] CRAN (R 4.0.2)                       
#>  prettyunits    1.1.1      2020-01-24 [1] CRAN (R 4.0.0)                       
#>  processx       3.4.4      2020-09-03 [1] CRAN (R 4.0.2)                       
#>  ps             1.3.4      2020-08-11 [1] CRAN (R 4.0.2)                       
#>  purrr          0.3.4      2020-04-17 [1] CRAN (R 4.0.0)                       
#>  R6             2.4.1      2019-11-12 [1] CRAN (R 4.0.0)                       
#>  ranger         0.12.1     2020-01-10 [1] CRAN (R 4.0.0)                       
#>  Rcpp           1.0.5      2020-07-06 [1] CRAN (R 4.0.0)                       
#>  remotes        2.1.1      2020-02-15 [1] CRAN (R 4.0.0)                       
#>  rlang          0.4.7      2020-07-09 [1] CRAN (R 4.0.2)                       
#>  rmarkdown      2.3        2020-06-18 [1] CRAN (R 4.0.2)                       
#>  rpart          4.1-15     2019-04-12 [1] CRAN (R 4.0.2)                       
#>  rprojroot      1.3-2      2018-01-03 [1] CRAN (R 4.0.0)                       
#>  scales         1.1.1      2020-05-11 [1] CRAN (R 4.0.2)                       
#>  sessioninfo    1.1.1      2018-11-05 [1] CRAN (R 4.0.0)                       
#>  stringi        1.4.6      2020-02-17 [1] CRAN (R 4.0.0)                       
#>  stringr        1.4.0      2019-02-10 [1] CRAN (R 4.0.0)                       
#>  testthat       2.3.2      2020-03-02 [1] CRAN (R 4.0.0)                       
#>  tibble         3.0.3      2020-07-10 [1] CRAN (R 4.0.2)                       
#>  tidyselect     1.1.0      2020-05-11 [1] CRAN (R 4.0.2)                       
#>  usethis        1.6.0      2020-04-09 [1] CRAN (R 4.0.0)                       
#>  uuid           0.1-4      2020-02-26 [1] CRAN (R 4.0.0)                       
#>  vctrs          0.3.4      2020-08-29 [1] CRAN (R 4.0.2)                       
#>  withr          2.2.0      2020-04-20 [1] CRAN (R 4.0.0)                       
#>  xfun           0.16       2020-07-24 [1] CRAN (R 4.0.2)                       
#>  xml2           1.3.1      2020-04-09 [1] CRAN (R 4.0.0)                       
#>  yaml           2.2.1      2020-02-01 [1] CRAN (R 4.0.0)                       
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

**Comment by jannes-m** [comment]
I have observed the same bug with my data. Here, I only add a plot using the correct labeling for better comparison (building on your example @RaphaelS1). 

``` r
library(mlr3learners)
library(mlr3viz)
library(mlr3)
library(cowplot)
library(ggplot2)

task = tsk("boston_housing")

learners = lrns(c("regr.featureless", "regr.rpart", "regr.ranger"))
measure = msr("regr.rmse")
set.seed(1)
bmr = benchmark(benchmark_grid(task, learners, rsmp("cv", folds = 3)))
#> INFO  [09:36:42.533] Benchmark with 9 resampling iterations 
#> INFO  [09:36:42.645] Applying learner 'regr.featureless' on task 'boston_housing' (iter 1/3) 
#> INFO  [09:36:42.661] Applying learner 'regr.featureless' on task 'boston_housing' (iter 2/3) 
#> INFO  [09:36:42.668] Applying learner 'regr.featureless' on task 'boston_housing' (iter 3/3) 
#> INFO  [09:36:42.674] Applying learner 'regr.rpart' on task 'boston_housing' (iter 1/3) 
#> INFO  [09:36:42.707] Applying learner 'regr.rpart' on task 'boston_housing' (iter 2/3) 
#> INFO  [09:36:42.722] Applying learner 'regr.rpart' on task 'boston_housing' (iter 3/3) 
#> INFO  [09:36:42.736] Applying learner 'regr.ranger' on task 'boston_housing' (iter 1/3) 
#> INFO  [09:36:43.506] Applying learner 'regr.ranger' on task 'boston_housing' (iter 2/3) 
#> INFO  [09:36:43.573] Applying learner 'regr.ranger' on task 'boston_housing' (iter 3/3) 
#> INFO  [09:36:43.647] Finished benchmark
sc = bmr$score()
p_1 = mlr3viz::autoplot(bmr)
p_2 = ggplot(sc, aes(x = learner_id, y = regr.mse)) +
  geom_boxplot() + 
  xlab(label = "")
cowplot::plot_grid(p_1, p_2)
```

![](https://i.imgur.com/jzOjRCE.png)

<sup>Created on 2020-10-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value                       
#>  version  R version 4.0.2 (2020-06-22)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  ctype    English_United States.1252  
#>  tz       Europe/Berlin               
#>  date     2020-10-02                  
#> 
#> - Packages -------------------------------------------------------------------
#>  package      * version date       lib source        
#>  assertthat     0.2.1   2019-03-21 [1] CRAN (R 4.0.2)
#>  backports      1.1.10  2020-09-15 [1] CRAN (R 4.0.2)
#>  callr          3.4.4   2020-09-07 [1] CRAN (R 4.0.2)
#>  checkmate      2.0.0   2020-02-06 [1] CRAN (R 4.0.2)
#>  cli            2.0.2   2020-02-28 [1] CRAN (R 4.0.2)
#>  codetools      0.2-16  2018-12-24 [2] CRAN (R 4.0.2)
#>  colorspace     1.4-1   2019-03-18 [1] CRAN (R 4.0.2)
#>  cowplot      * 1.1.0   2020-09-08 [1] CRAN (R 4.0.2)
#>  crayon         1.3.4   2017-09-16 [1] CRAN (R 4.0.2)
#>  curl           4.3     2019-12-02 [1] CRAN (R 4.0.2)
#>  data.table     1.13.0  2020-07-24 [1] CRAN (R 4.0.2)
#>  desc           1.2.0   2018-05-01 [1] CRAN (R 4.0.2)
#>  devtools       2.3.2   2020-09-18 [1] CRAN (R 4.0.2)
#>  digest         0.6.25  2020-02-23 [1] CRAN (R 4.0.2)
#>  dplyr          1.0.2   2020-08-18 [1] CRAN (R 4.0.2)
#>  ellipsis       0.3.1   2020-05-15 [1] CRAN (R 4.0.2)
#>  evaluate       0.14    2019-05-28 [1] CRAN (R 4.0.2)
#>  fansi          0.4.1   2020-01-08 [1] CRAN (R 4.0.2)
#>  farver         2.0.3   2020-01-16 [1] CRAN (R 4.0.2)
#>  fs             1.5.0   2020-07-31 [1] CRAN (R 4.0.2)
#>  future         1.19.1  2020-09-22 [1] CRAN (R 4.0.2)
#>  future.apply   1.6.0   2020-07-01 [1] CRAN (R 4.0.2)
#>  generics       0.0.2   2018-11-29 [1] CRAN (R 4.0.2)
#>  ggplot2      * 3.3.2   2020-06-19 [1] CRAN (R 4.0.2)
#>  globals        0.13.0  2020-09-17 [1] CRAN (R 4.0.2)
#>  glue           1.4.2   2020-08-27 [1] CRAN (R 4.0.2)
#>  gtable         0.3.0   2019-03-25 [1] CRAN (R 4.0.2)
#>  highr          0.8     2019-03-20 [1] CRAN (R 4.0.2)
#>  htmltools      0.5.0   2020-06-16 [1] CRAN (R 4.0.2)
#>  httr           1.4.2   2020-07-20 [1] CRAN (R 4.0.2)
#>  knitr          1.30    2020-09-22 [1] CRAN (R 4.0.2)
#>  labeling       0.3     2014-08-23 [1] CRAN (R 4.0.0)
#>  lattice        0.20-41 2020-04-02 [2] CRAN (R 4.0.2)
#>  lgr            0.3.4   2020-03-20 [1] CRAN (R 4.0.2)
#>  lifecycle      0.2.0   2020-03-06 [1] CRAN (R 4.0.2)
#>  listenv        0.8.0   2019-12-05 [1] CRAN (R 4.0.2)
#>  magrittr       1.5     2014-11-22 [1] CRAN (R 4.0.2)
#>  Matrix         1.2-18  2019-11-27 [2] CRAN (R 4.0.2)
#>  memoise        1.1.0   2017-04-21 [1] CRAN (R 4.0.2)
#>  mime           0.9     2020-02-04 [1] CRAN (R 4.0.0)
#>  mlr3         * 0.6.0   2020-09-13 [1] CRAN (R 4.0.2)
#>  mlr3learners * 0.4.0   2020-09-25 [1] CRAN (R 4.0.2)
#>  mlr3measures   0.2.0   2020-06-27 [1] CRAN (R 4.0.2)
#>  mlr3misc       0.5.0   2020-08-13 [1] CRAN (R 4.0.2)
#>  mlr3viz      * 0.3.0   2020-09-25 [1] CRAN (R 4.0.2)
#>  munsell        0.5.0   2018-06-12 [1] CRAN (R 4.0.2)
#>  paradox        0.4.0   2020-07-21 [1] CRAN (R 4.0.2)
#>  pillar         1.4.6   2020-07-10 [1] CRAN (R 4.0.2)
#>  pkgbuild       1.1.0   2020-07-13 [1] CRAN (R 4.0.2)
#>  pkgconfig      2.0.3   2019-09-22 [1] CRAN (R 4.0.2)
#>  pkgload        1.1.0   2020-05-29 [1] CRAN (R 4.0.2)
#>  prettyunits    1.1.1   2020-01-24 [1] CRAN (R 4.0.2)
#>  processx       3.4.4   2020-09-03 [1] CRAN (R 4.0.2)
#>  ps             1.3.4   2020-08-11 [1] CRAN (R 4.0.2)
#>  purrr          0.3.4   2020-04-17 [1] CRAN (R 4.0.2)
#>  R6             2.4.1   2019-11-12 [1] CRAN (R 4.0.2)
#>  ranger         0.12.1  2020-01-10 [1] CRAN (R 4.0.2)
#>  Rcpp           1.0.5   2020-07-06 [1] CRAN (R 4.0.2)
#>  remotes        2.2.0   2020-07-21 [1] CRAN (R 4.0.2)
#>  rlang          0.4.7   2020-07-09 [1] CRAN (R 4.0.2)
#>  rmarkdown      2.3     2020-06-18 [1] CRAN (R 4.0.2)
#>  rpart          4.1-15  2019-04-12 [2] CRAN (R 4.0.2)
#>  rprojroot      1.3-2   2018-01-03 [1] CRAN (R 4.0.2)
#>  scales         1.1.1   2020-05-11 [1] CRAN (R 4.0.2)
#>  sessioninfo    1.1.1   2018-11-05 [1] CRAN (R 4.0.2)
#>  stringi        1.5.3   2020-09-09 [1] CRAN (R 4.0.2)
#>  stringr        1.4.0   2019-02-10 [1] CRAN (R 4.0.2)
#>  testthat       2.3.2   2020-03-02 [1] CRAN (R 4.0.2)
#>  tibble         3.0.3   2020-07-10 [1] CRAN (R 4.0.2)
#>  tidyselect     1.1.0   2020-05-11 [1] CRAN (R 4.0.2)
#>  usethis        1.6.3   2020-09-17 [1] CRAN (R 4.0.2)
#>  uuid           0.1-4   2020-02-26 [1] CRAN (R 4.0.0)
#>  vctrs          0.3.4   2020-08-29 [1] CRAN (R 4.0.2)
#>  withr          2.3.0   2020-09-22 [1] CRAN (R 4.0.2)
#>  xfun           0.17    2020-09-09 [1] CRAN (R 4.0.2)
#>  xml2           1.3.2   2020-04-23 [1] CRAN (R 4.0.2)
#>  yaml           2.2.1   2020-02-01 [1] CRAN (R 4.0.2)
#> 
#> [1] C:/Users/jannes.muenchow/R/win-library/4.0
#> [2] C:/Program Files/R/R-4.0.2/library
```

</details>

**Comment by mllg** [comment]
Confirmed. Introduced via 86f5f129.

**Comment by mllg** [comment]
Fixed in dfa15d41a453bc67c23b39f0687aae33e312f2ae.
